patches/tModLoader/Terraria/ModLoader/ModBiome.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Generic;
+using Terraria.WorldBuilding;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// This class represents a biome added by a mod. It exists to centralize various biome related hooks, handling a lot of biome boilerplate. Use its various logic hooks act as if they were in ModPlayer, and use ModifyWorldGenTasks as if it were in ModWorld.
+	/// </summary>
+	public abstract class ModBiome : ModType
+	{
+		/// <summary>
+		/// The player associated with this ModBiome object. This does not relate to the local player in any way.
+		/// </summary>
+		public Player player { get; internal set; }
+
+		internal int index;
+
+		/// <summary>
+		/// Whether or not the player is in this specific biome.
+		/// </summary>
+		public bool Active { get; internal set; }
+
+		protected override void Register() {
+			ModTypeLookup<ModBiome>.Register(this);
+			BiomeLoader.Add(this);
+		}
+
+		internal void Update() {
```

Comment:
This concept of `Active` and the internalized `Update` method differ from I think all current hook types if I'm not mistaken, but it was the best I could come up with. I'm open to suggestions for changes.